src/math/vec2.js

```diff
@@ -8,17 +8,45 @@
  * @example
  * var v = new pc.Vec2(1, 2);
  */
-function Vec2(x, y) {
-    if (x && x.length === 2) {
-        this.x = x[0];
-        this.y = x[1];
-    } else {
-        this.x = x || 0;
-        this.y = y || 0;
+/**
+ * @field
+ * @name pc.Vec2#x
+ * @type {number}
+ * @description The first element of the vector.
+ * @example
+ * var vec = new pc.Vec2(10, 20);
+ *
+ * // Get x
+ * var x = vec.x;
+ *
+ * // Set x
+ * vec.x = 0;
+ */
+/**
+ * @field
+ * @name pc.Vec2#y
+ * @type {number}
+ * @description The second element of the vector.
+ * @example
+ * var vec = new pc.Vec2(10, 20);
+ *
+ * // Get y
+ * var y = vec.y;
+ *
+ * // Set y
+ * vec.y = 0;
+ */
+class Vec2 {
+    constructor(x = 0, y = 0) {
+        if (x && x.length === 2) {

```

Comment:
can be changed to 
`if (x.length === 2) {`

Reply:
Fixed